frame/assets/src/types.rs

```diff
@@ -83,45 +89,57 @@ pub struct Approval<Balance, DepositBalance> {
 
 #[test]
 fn ensure_bool_decodes_to_consumer_or_sufficient() {
-	assert_eq!(false.encode(), ExistenceReason::<()>::Consumer.encode());
-	assert_eq!(true.encode(), ExistenceReason::<()>::Sufficient.encode());
+	assert_eq!(false.encode(), ExistenceReason::<(), ()>::Consumer.encode());
+	assert_eq!(true.encode(), ExistenceReason::<(), ()>::Sufficient.encode());
 }
 
+/// The reason for an account's existence within an asset class.
 #[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, MaxEncodedLen, TypeInfo)]
-pub enum ExistenceReason<Balance> {
+pub enum ExistenceReason<Balance, AccountId> {
+	/// A consumer reference was used to create this account.
 	#[codec(index = 0)]
 	Consumer,
+	/// The asset class is `sufficient` for account existence.
 	#[codec(index = 1)]
 	Sufficient,
+	/// The `AccountId` has placed a deposit to exist within an asset class.
```

Comment:
`AccountId` is just generic argument in this context, its not clear from this that deposit taken from the owner of the account.